megablocks/layers/router.py

```diff
@@ -43,21 +43,16 @@ def jitter(self, x):
         noise = torch.rand(x.size(), dtype=x.dtype, device=x.device)
         return low + noise * (high - low)
 
-    def _top_k(self, scores):
-        if self.args.moe_top_k == 1:
-            return scores.max(dim=-1)
```

Comment:
Rather than delete this optimization, could you `unsqueeze` the last dimension in the `max` case to fix this bug?

Reply:
fixed by simply using [keepdim=True](https://pytorch.org/docs/stable/generated/torch.Tensor.max.html) instead